### ISSUE TITLE
I would like to tell you why it might not still be able to run valorant even after so many tweaking 

### ISSUE THREAD

**Issue by Greatz08** [zhaodice/qemu-anti-detection#96]
Valorant requires TPM 2.0 and secure boot BOTH enabled otherwise it wont let you run even if you are running in raw windows (not virtualized) . So that might be the reason for it but i guess it would show error to enable secure boot and tpm maybe they are detecting CIK values first because of which you are not getting to the point where it would detect these two things and show those errors ?

**Comment by zhaodice** [comment]
Windows 10 can bypass TPM detection, but it is not a good idea

**Comment by Greatz08** [comment]
@zhaodice you can passthrough tpm too so tpm is not something to worry about about but secure boot is issue because in q35 chipset with UEFI i tried to install windows 11 with tpm passthrough still it didnt enabled secure boot by default so having issue with that aspect so can you try to create guide for it ?
Plus those CIK values i tried to study on that are related to ACPI tables. We have to decompile it and then modify dsdt file in such a way that it has fake CIK values which i know will be static but still its better than having no value. Kernel level anticheats can still detect and see if value is changing or not and everyone knows that values of sensors data have to change at regular interval of time so its still not the best solution but still it will bypass atleast al-khaseer script detection . 
Wish their was dynamic way of passing the values from main system to qemu machine it would solve alot of issues related to sensors data

**Comment by zhaodice** [comment]
> @zhaodice you can passthrough tpm too so tpm is not something to worry about about but secure boot is issue because in q35 chipset with UEFI i tried to install windows 11 with tpm passthrough still it didnt enabled secure boot by default so having issue with that aspect so can you try to create guide for it ? Plus those CIK values i tried to study on that are related to ACPI tables. We have to decompile it and then modify dsdt file in such a way that it has fake CIK values which i know will be static but still its better than having no value. Kernel level anticheats can still detect and see if value is changing or not and everyone knows that values of sensors data have to change at regular interval of time so its still not the best solution but still it will bypass atleast al-khaseer script detection . Wish their was dynamic way of passing the values from main system to qemu machine it would solve alot of issues related to sensors data

1.I recall that PROXMOX's UEFI OVMF is capable of using secure boot, but OVMF is obviously a virtualization component and you may have to remove the OVMF features from the source code yourself. Compiling the OVMF yourself is another huge project

2.If you need a dynamic way to pass this value, you will most likely need KVM, if the KVM driver can create a local /dev/virtual device file, through the echo can adjust the sensor value, it will be a big breakthrough

**Comment by Kojima648** [comment]
Do you mean you can pass valorant's detection of TPM2.0 using Win10? Which version exactly?

**Comment by Greatz08** [comment]
@zhaodice yeah in short it is too much work to fix those things but if someone is able to accomplish these two big things in windows 11, then it might create some chance to bypass valorant detection. 

We wont ever get any support regarding this from qemu devs or kernel devs so the person who can understand qemu and kvm and can think of a solution for those mentioned issues , That person can only become ray of hope for gamers as max game are creating anticheats to accomplish same level of security like valorant just that they dont have enough capable people or maybe they do have but they have to implement anticheat at kernel level which can create negative impact so maybe that thing is holding them back idk but the length valorant is going through to stop us from playing in qemu is crazyyy . I know exact reasons for why they did but still running anticheat at kernel level is not something that anyone would have thought of because running anything at kernel can become big matter of concern.

**Comment by chenbowen9706** [comment]
actually,only modify qemu you will never bypass vanguard,take more time to look msr and cpuid,there   are too many decetions.eg:lstar,arch lbr.by the way you dont need to fake rdtsc or rdtscp time if you turn on nested guest hyperv
